datastore/api/snippets.py

```diff
@@ -0,0 +1,823 @@
+# Copyright 2016, Google, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import datetime
+from pprint import pprint
+import time
+
+import gcloud
+from gcloud import datastore
+
+
+def incomplete_key(client):
+    # [START incomplete_key]
+    key = client.key('Task')
+    # [END incomplete_key]
+
+    return key
+
+
+def named_key(client):
+    # [START named_key]
+    key = client.key('Task', 'sample_task')
+    # [END named_key]
+
+    return key
+
+
+def key_with_parent(client):
+    # [START key_with_parent]
+    key = client.key('TaskList', 'default', 'Task', 'sample_task')
+    # Alternatively
+    parent_key = client.key('TaskList', 'default')
+    key = client.key('Task', 'sample_task', parent=parent_key)
+    # [END key_with_parent]
+
+    return key
+
+
+def key_with_multilevel_parent(client):
+    # [START key_with_multilevel_parent]
+    key = client.key(
+        'User', 'alice',
+        'TaskList', 'default',
+        'Task', 'sample_task')
+    # [END key_with_multilevel_parent]
+
+    return key
+
+
+def basic_entity(client):
+    # [START basic_entity]
+    task = datastore.Entity(client.key('Task'))
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+    # [END basic_entity]
+
+    return task
+
+
+def entity_with_parent(client):
+    # [START entity_with_parent]
+    key_with_parent = client.key(
+        'TaskList', 'default', 'Task', 'sample_task')
+
+    task = datastore.Entity(key=key_with_parent)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+    # [END entity_with_parent]
+
+    return task
+
+
+def properties(client):
+    # [START properties]
+    task = datastore.Entity()
+    task.update({
+        # String
```

Comment:
I would rather have the docs state (but better worded): This snippet creates a string property called 'type', a date property called 'created', etc).